tests/dialects/test_postgres.py

```diff
@@ -819,6 +815,13 @@ def test_postgres(self):
             },
         )
 
+        self.validate_all(
+            "/*+ some comment*/ SELECT b.foo, b.bar FROM baz AS b",
+            write={
+                "postgres": UnsupportedError,
+            },
+        )
```

Comment:
Why is this unsupported? This change doesn't look right, this isn't really a hint, right?

Reply:
If we add the hint tokens in `COMMENTS`, then this will be picked up as an `exp.Hint` in the respective dialect (e.g Postgres). If it's not supported, the roundtrip will yield unsupported through the generator flag `QUERY_HINTS`.

Reply:
Ok, this looks like an issue. I think for comments to be treated as hints, the previous token needs to be a `SELECT`. That's how it worked until today:

```python
        if self._match(TokenType.SELECT):
            comments = self._prev_comments

            hint = self._parse_hint()
```

Also, you need to make sure to _only_ use the last comment for a hint. Otherwise if there are actual comments that would be attached to the select, you don't want to treat them as hints. I'm not sure if you can achieve that somehow, but I'd test this more:

```

SELECT /* some ordinary comment */ /*+ some hint */ ...

SELECT /* some ordinary comment */
  /*+ some hint */
  ...

etc
```

Reply:
Regarding the hint placement, I think this is right, but I guess the intention of this case was to test hint tokenization/parsing for dialects that do not support it e.g in Postgres, this would be treated as a regular comment so it could be placed _anywhere_, whereas now that's not the case. Is this an issue? Looks like an extreme edge case.

About the comments, I'll do more testing to make sure but I thought it wouldn't be an issue; the tokenizer's `scan_comment` will scan just the hint and attach it to the produced `TokenType.HINT`, so in general we should expect only 1 attached comment and that's on the hint token/expression

Reply:
> Is this an issue? Looks like an extreme edge case.

Yeah, let's try to do this correctly without making assumptions and breaking valid queries:

```
georgesittas=# /*+ some comment */ select 1;
 ?column?
----------
        1
(1 row)
```

We should just preserve the previous behavior and only treat hint comments as hints if they're right after the `SELECT` token.